### PR TITLE
Use atomic-openshift-node to determine node roles for tuned.

### PR DESCRIPTION
**Pull request by jmencak**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1612776

### CODE REVIEW

File: roles/openshift_node_group/tasks/bootstrap_config.yml [hunk 1]
```diff
@@ -15,6 +15,9 @@
     line: "BOOTSTRAP_CONFIG_NAME={{ openshift_node_group_name }}"
     regexp: "^BOOTSTRAP_CONFIG_NAME=.*"
 
+- import_role:
```
Comment (vrutkovs) [inline roles/openshift_node_group/tasks/bootstrap_config.yml]: Not sure why is this required - why not leaving the call in `playbooks/openshift-node/private/configure_nodes.yml`?

File: roles/tuned/templates/recommend.conf [hunk 2]
```diff
@@ -1,11 +1,11 @@
 [openshift-control-plane,master]
-/etc/origin/master/master-config.yaml=.*
+/etc/sysconfig/atomic-openshift-node=.*BOOTSTRAP_CONFIG_NAME.*master
 
-[openshift-control-plane,node]
-/etc/origin/node/node-config.yaml=.*node-role.kubernetes.io/infra=true
+[openshift-control-plane,infra]
+/etc/sysconfig/atomic-openshift-node=.*BOOTSTRAP_CONFIG_NAME.*infra
```
Comment (vrutkovs) [inline roles/tuned/templates/recommend.conf]: Why we wouldn't want to apply this to Origin too?

File: roles/tuned/templates/recommend.conf [hunk 3]
```diff
@@ -1,11 +1,11 @@
 [openshift-control-plane,master]
-/etc/origin/master/master-config.yaml=.*
+/etc/sysconfig/atomic-openshift-node=.*BOOTSTRAP_CONFIG_NAME.*master
```
Comment (vrutkovs) [inline roles/tuned/templates/recommend.conf]: `BOOTSTRAP_CONFIG_NAME` is not guarenteed to be present in sysconfig settings - its gone after 3.10 -> 3.11 upgrade for instance

File: roles/openshift_node_group/tasks/bootstrap_config.yml (hunk 1 above)
Comment (vrutkovs) [reply roles/openshift_node_group/tasks/bootstrap_config.yml]: Ah, okay, makes sense

File: roles/tuned/templates/recommend.conf (hunk 3 above)
Comment (vrutkovs) [reply roles/tuned/templates/recommend.conf]: Why not leaving just `node-config.yaml`? BOOTSTRAP_CONFIG_NAME is used just once during initial setup and isn't applicable during node run - its would use `node-config.yaml`. sysconfig would be reset during 3.10 -> 3.11 upgrade and won't have any config mentioned.

The original issue has been filed earlier than master/infra/compute were written into node-config, but currently it would certainly be written there. Also, `/etc/origin/node/node-config.yaml` would appear both in OKD and OCP - so its seems like a natural choice to me

Comment (jmencak) [reply roles/tuned/templates/recommend.conf]: Because node-config.yaml is not populated by the node labels long after the "Node Bootstrap Preparation" is complete.  Even tasks like GlusterFS install start before the labels are set in node-config.yaml.  That's why I moved the location I call the "tuned" role and used atomic-openshift-node, as this is the file the node role is visible first.  I'd love to use node-config.yaml exclusively for this (see https://bugzilla.redhat.com/show_bug.cgi?id=1569917), but it looks this is simply not possible at present with major changes.  Any other suggestions for fixing this cleanly and quickly welcome.  In the future, the planned fix will be containerized tuned watching node labels changes.  Thanks for the review, btw!

**Comment by DanyC97** [comment]
@jmckind is interesting because in 3.7 origin i do see both master/ nodes having the same settings. In 1.5/ 3.5 it was like you just mentioned here.

I know/ understand i'm way off from what you are testing now but i thought i should just make the note...

File: roles/tuned/templates/recommend.conf (hunk 3 above)
Comment (sdodson) [reply roles/tuned/templates/recommend.conf]: We also cannot rely on the configmap name matching these strings. They match in our defaults, but it's perfectly valid to name your configmap `node-config-sdodson` and that configmap applies master labels.

**Comment by jmencak** [comment]
Closing in favour of https://github.com/openshift/openshift-ansible/pull/10102
